### PR TITLE
Feat/135 navbar component refactor

### LINKED ISSUE #135: Navbar component refactor
Navbar component needs refactored, needs tested on IphoneXR and Desktop screen sizes
[Figma Design Link](https://www.figma.com/file/rs7dsdmKlxdfuQ0wobp5B2/Data-Umbrella?node-id=220%3A462)

- [•] Navbar Menu Item bolds on hover, when it should have same font weight
- [•] need to replace 'Sponsors' link with 'Support Us' dropdown menu button
- [•] add more L/R padding to navbar links/buttons
- [•] dark mode nav links need to be aqua color on hover/white otherwise
- [•] add hover state to all elements
- [•] Mobile navbar needs to switch to hamburger menu w/ more options on mobile
- [•] dark/light mode option not present on mobile navbar
- [•] sun/moon icon not being toggled consistently for light/dark mode
- [•] change Data Umbrella logo/text in dark mode
- [•] change Data Umbrella logo to scale logo in mobile

### PR DESCRIPTION
**Pull request by Saige243**
Fixes #135 

- [x] Navbar Menu Item bolds on hover, when it should have same font weight
- [x] need to replace 'Sponsors' link with 'Support Us' dropdown menu button
- [x] add more L/R padding to navbar links/buttons
- [x] dark mode nav links need to be aqua color on hover/white otherwise
- [x] add hover state to all elements
- [x] Mobile navbar needs to switch to hamburger menu w/ more options on mobile
- [x] dark/light mode option not present on mobile navbar
- [x] sun/moon icon not being toggled consistently for light/dark mode
- [x] change Data Umbrella logo/text in dark mode
- [x] change Data Umbrella logo to scale logo in mobile

## Before and After for UI Updates

Before:
Screen Shot 2022-09-12 at 8 32 34 AM

After:
Screen Shot 2022-09-12 at 9 11 24 AM

(No hover state on mobile, so no screenshots there.)

## [694cc78](https://github.com/data-umbrella/event-board-web/pull/165/commits/694cc78f84057f8b09d82f83a68d3f3cd7dc34e7)

Fixed spacing issue on navbar elements.
Before:
Screen Shot 2022-09-12 at 8 32 34 AM

After:
Screen Shot 2022-09-12 at 9 39 58 AM

(Diff navbar on mobile, so no screenshots there.)

## [94bec2e](https://github.com/data-umbrella/event-board-web/pull/165/commits/94bec2ee0466c5980a0ab8a2f7096a976760813a)

Corrected the color on navbar links hover:
Before (plain white):
Screen Shot 2022-09-12 at 10 37 12 AM

After:
Screen Shot 2022-09-12 at 10 36 05 AM

## [ddf1c8d](https://github.com/data-umbrella/event-board-web/pull/165/commits/ddf1c8d1b39793356b8062d5f49f783b54443377)

Added hover and underline effect on events dropdown per figma:
Before:
Screen Shot 2022-09-12 at 10 38 52 AM

After:
Screen Shot 2022-09-12 at 10 42 05 AM

## [ea1b015](https://github.com/data-umbrella/event-board-web/pull/165/commits/ea1b015aed2e213bf0c1ddd05c4f8523b06f7d55)
Changed light/dark icon for theme on desktop.

Before:
Screen Shot 2022-09-12 at 10 38 52 AM
(moon still showing though it should be the sun.)

After:
Screen Shot 2022-09-12 at 12 09 40 PM
(sun now showing for toggling light mode)

## [5a16525](https://github.com/data-umbrella/event-board-web/pull/165/commits/5a16525483f292472e1ba902ebb06bb32fc58ff0)

Before (dark umbrella):
Screen Shot 2022-09-12 at 12 09 40 PM

After (light umbrella):
Screen Shot 2022-09-12 at 12 23 59 PM

Before (no mobile nav):
Screen Shot 2022-09-13 at 10 43 08

After (with dark/light mode option included):
Screen Shot 2022-09-13 at 10 43 40

Screen Shot 2022-09-13 at 10 43 43

Screen Shot 2022-09-13 at 10 43 48

Screen Shot 2022-09-13 at 10 48 05

## For PR Reviewer
- [ ] Does this file change the yarn.lock, package.json or package-lock.json file? If so, why?
- [ ] If this pr contains mobile and desktop changes, did you test on IphoneXr and desktop views?
- [ ] Does this file match the related tickets linked figma file, or does it pass the visual smell test?
- [ ] If this file contains javascript, does the javascript pass the smell test?
- [ ] If you don't feel super confident in your review, did you assign someone more senior to double check?

### CODE REVIEW

File: src/components/elements/MobileNav.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+function MobileNav() {
+  const [menuOpen, setMenuOpen] = useState(false)
+  const [colorTheme, setTheme] = useDarkMode();
+
+
+  return (
+    <>
+      <div className='grid text-right bg-du-gray dark:bg-slate-700 h-16 grid-cols-4 place-content-center place-items-center'>
```
Comment (dashcraft) [inline src/components/elements/MobileNav.jsx:20]: comment (blocking):
class naming should be in the order of...

`positioning padding/margins colors other things`

your order right now has text-right next to grid, that needs to be on the other side. Might need to use justify-content or align content

File: src/components/elements/MobileNav.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <section className='place-content-center place-items-center'>
+          <Link onClick={() => setMenuOpen(false)} to="/"><img className="object-scale-down object-center h-14 w-16" src={colorTheme === 'dark' ? Logo : logoDark} alt="logo" /></Link>
+        </section>
+        <span className='text-center col-span-2'>
```
Comment (dashcraft) [inline src/components/elements/MobileNav.jsx:24]: switch class order

File: src/components/elements/MobileNav.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            </button>
+            <div>
+              <img
+                src={colorTheme === 'dark' ? darkMode : lightMode}
+                onClick={() => setTheme(colorTheme)}
+                className="float-right pb-2"
+                alt="logo"
+              />
+            </div>
+            <Link onClick={() => setMenuOpen((prev) => !prev)} to="/sign-in">Sign In</Link><br />
```
Comment (dashcraft) [inline src/components/elements/MobileNav.jsx:45]: setMenuOpen with arguments gets copied and pasted a lot, create a handler like...

function handleToggleMenu(e) {
  e.preventDefault()
  setMenuOpen(prev => !prev)
}

then...

onClick={handleToggleMenu}

File: tailwind.config.js [hunk 4]
```diff
@@ -13,7 +13,9 @@ module.exports = {
           600: '#783D86',
         },
         'du-gray': '#F7FAFC',
-      }
+        'lightAqua': '#00FFFF',
```
Comment (dashcraft) [inline tailwind.config.js:16]: use `du-lightAqua` so we don't accidentally override tailwind styles

**Review by dashcraft — CHANGES_REQUESTED** [review]
quick change requests

File: src/components/elements/MobileNav.jsx (hunk 3 above)
Comment (Saige243) [reply src/components/elements/MobileNav.jsx:45]: Imported useEffect w/ useLocation()

**Pull request merged**